server/src/main/java/org/elasticsearch/search/aggregations/bucket/missing/MissingAggregator.java

```diff
@@ -49,17 +49,16 @@ public MissingAggregator(String name, AggregatorFactories factories, ValuesSourc
     @Override
     public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             final LeafBucketCollector sub) throws IOException {
-
-        final Bits docsWithValue;
+        final DocValueBits docsWithValue;
         if (valuesSource != null) {
             docsWithValue = valuesSource.docsWithValue(ctx);
         } else {
-            docsWithValue = new Bits.MatchNoBits(ctx.reader().maxDoc());
+            docsWithValue = null;
```

Comment:
I don't think this is correct? Do tests pass? This should fail on unmapped fields.

Reply:
Sorry for that. I thought ```gradle test``` will run all tests. I pushed 6904e0f5e7c2ef0bf4a417a3196c60e611bb4337